File: Data Structure/Array or Matrix/Medium/LET_no238_Product of Array Except Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int product = 1, n = nums.size();
        vector<int> res(n);
        for (int i = 0; i < n; i++) {
            res[i] = product;
            product *= nums[i];
        }
        product = 1;
        for (int i = n - 1; i >= 0; i--) {
            res[i] *= product;
            product *= nums[i];
        }
        return res;
    }
};

File: Data Structure/Array or Matrix/Medium/LET_no238_Product of Array Except Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = []
        product = 1
        for i in range(n):
            res.append(product)
            product *= nums[i]
        product = 1
        for i in range(n - 1, -1, -1):
            res[i] *= product
            product *= nums[i]
        return res